presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -349,22 +349,44 @@ public void testJoinPartitionedWithMissingPartitionFilter()
     @Test
     public void testNaNPartition()
     {
+        // Only NaN partition
         assertUpdate("DROP TABLE IF EXISTS test_nan_partition");
         assertUpdate("CREATE TABLE test_nan_partition(a varchar, d double) WITH (partitioned_by = ARRAY['d'])");
+        assertUpdate("INSERT INTO test_nan_partition VALUES ('b', nan())", 1);
```

Comment:
let's have a new dedicated test, instead of making this one more complicate

Reply:
i thought about this, but i couldn't invent the name for the test method.
Also, this is the test which was supposed to test the problem (d068a3f5e656118dcb4ca0d26877b0359c6b5a2c) and indeed it did, but not deterministically (order of partitions on the list mattered). So I am improving the existing test to make it more deterministic.

Reply:
Discussed offline.